src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs

```diff
@@ -72,80 +78,12 @@ private SourceText CreateTextInternal(Stream stream, Encoding encoding, Cancella
             cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Can you comment this function?  I find it confusing that an Encoding is passed in, but then the code calls "detectEncodingFromBytOrderMarks" and then calls AsRoslynText, not using the encoding passed in, but instead reader.CurrentEncoding or Encoding.UTF8.

It seems like someone can be explicit about the encoding, but still have that overridden by the system.